applications/MultilevelMonteCarloLaplacianApplication/python_scripts/pure_diffusion_solver.py

```diff
@@ -0,0 +1,241 @@
+        self.builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(self.linear_solver)
+
+        # self.solver = KratosMultiphysics.ResidualBasedLinearStrategy(self.main_model_part,
+        #                                                              self.time_scheme,
+        #                                                              self.linear_solver,
+        #                                                              self.builder_and_solver,
+        #                                                              self.settings["compute_reactions"].GetBool(),
+        #                                                              self.settings["reform_dofs_at_each_step"].GetBool(),
+        #                                                              self.settings["calculate_norm_dx"].GetBool(),
+        #                                                              self.settings["move_mesh_flag"].GetBool())
+
+        self.solver = KratosMultiphysics.ResidualBasedLinearStrategy(self.main_model_part,
+                                                                     self.time_scheme,
+                                                                     self.linear_solver,
+                                                                     self.builder_and_solver,
+                                                                     False,
+                                                                     False,
+                                                                     False,
+                                                                     False)
+
+
+        (self.solver).SetEchoLevel(self.settings["echo_level"].GetInt())
+        (self.solver).Initialize()
+        (self.solver).Check()
+        # print ("Pure diffusion solver initialization finished")
+
+        
+    def Solve(self):
+        # Solve equations on mesh
+        print("!!!WARNING: YOU SHOULD NOT USE solver.Solve()!!!")
+        (self.solver).Solve()
+
+        
+    def SolveSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            is_converged = self.solver.SolveSolutionStep()
+            if not is_converged and self._IsPrintingRank():
+                msg  = "Pure Diffusion solver did not converge for step " + str(self.main_model_part.ProcessInfo[KratosMultiphysics.STEP]) + "\n"
+                msg += "corresponding to time " + str(self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]) + "\n"
+                KratosMultiphysics.Logger.PrintWarning("PureDiffusionSolver",msg)
+
+        
+    def Finalize(self):
+        # now we proceed to use the GID interface (both to import the infomation inside the .mdpa file and later print the results in a file
+        # since in the .json file the output configuration was not defined, we define it now
+        gid_mode = KratosMultiphysics.GiDPostMode.GiD_PostAscii  #we import the python file that includes the commands that we need
```

Comment:
Instead of doing this manually you could use the GiDOutputProcess from the core

Reply:
+1 for Philipp's comment

Reply:
Ok, I modified the json file to avoid to do it manually, thanks for the hint.